File: лаб№1 в/ньютона.cpp
#define _USE_MATH_DEFINES
#include <iostream>
#include <iomanip>
#include <cmath>
using namespace std;

double func(double x)
{
    return pow(M_E, x) * x - x - 1;//pow(M_E, x) вычисляет значение числа e (основание натурального логарифма) в степени x
}

double dfunc(double x)
{
    return pow(M_E, x) * (x + 1);
}

double find(double x, double eps)
{
    double f, df; 
    int iter = 0;
    cout << "x_0 = " << x << " ";
    do {
        f = func(x);
        df = dfunc(x);
        x = x - f / df;
        iter++;
    } while (fabs(f) > eps && iter < 20000);
    cout << "\nNumber of iterations is " << iter << endl;
    return x;
}

int main()
{
    double x0 = 0.0, x, eps = 1e-6;
    x = find(x0, eps);
    cout << "Root x = " << fixed << setprecision(16) << x;
    cout << "\nFunction value f(x) = " << func(x) << endl;

    return 0;
}

File: лаб№1 в/ньютона.py
import math

def func(x):
    return math.exp(x) * x - x - 1

def dfunc(x):
    return math.exp(x) * (x + 1)

def find(x, eps):
    iter = 0
    print("x_0 = ", x, end=" ")
    while True:
        f = func(x)
        df = dfunc(x)
        x = x - f / df
        iter += 1
        if abs(f) <= eps or iter >= 20000:
            break
    print("\nNumber of iterations is", iter)
    return x

def main():
    x0 = 0.0
    eps = 1e-6
    x = find(x0, eps)
    print("Root x =", format(x, '.16f'))
    print("Function value f(x) =", func(x))

if __name__ == "__main__":
    main()